routing/turns_tts_text.cpp

```diff
@@ -62,20 +297,150 @@ std::string GetTtsText::GetTurnNotification(Notification const & notification) c
   if (notification.m_useThenInsteadOfDistance && notification.m_turnDir == CarDirection::None)
     return {};
 
-  std::string const dirStr = GetTextById(GetDirectionTextId(notification));
   if (dirStr.empty())
     return {};
 
-  std::string distStr;
-  if (notification.m_useThenInsteadOfDistance)
-  {
-    distStr = GetTextById("then");
-    if (notification.m_distanceUnits > 0)
-      distStr = distStr + " " + GetTextById(GetDistanceTextId(notification));
+  std::string thenStr;
+  if (notification.m_useThenInsteadOfDistance) {
+    // add "then" and space only if needed, for appropriate languages
+    if(localeKey != "ja")
+      thenStr = GetTextById("then") + " ";
+    else
+      thenStr = GetTextById("then");
   }
-  else
+
+  std::string distStr;
+  if (notification.m_distanceUnits > 0)
     distStr = GetTextById(GetDistanceTextId(notification));
-  return distStr + " " + dirStr;
+
+  // Get a string like 245; CA 123; Highway 99; San Francisco
+  // In the future we could use the full RoadNameInfo struct to do some nice formatting.
+  std::string streetOut;
+  RouteSegment::RoadNameInfo nsi = notification.m_nextStreetInfo; // extract non-const
+  FormatFullRoadName(nsi, streetOut);
+
+  if (!streetOut.empty()) {
+    // We're going to pronounce the street name.
+
+    // Replace any full-stop characters (in between sub-instructions) to make TTS flow better.
```

Comment:
Should dots be removed from translations instead?

Reply:
It's possible however it's not really important and if there is some TTS engine out there that prefers sentences to be ended with hard stops (or a translator gets a little too eager) I don't want that to sound weird here. I erred on the side of a minimal change instead of presumptively redoing lots of stuff.